### PR TITLE
Chronos: add `from_prometheus` to initialize tsdataset(s) from Prometheus data

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Provide with new API to initialize tsdataset(s) from Prometheus data for specified time period via url.

### 1. Why the change?

https://github.com/intel-analytics/BigDL/issues/5896

### 2. User API changes

```python
def from_prometheus(prometheus_url, query, starttime, endtime, step,
                    target_col=None, id_col=None, extra_feature_col=None,
                    with_split=False, val_ratio=0, test_ratio=0.1, **kwargs):
```

Example:
```python
tsdata = TSDataset.from_prometheus(prometheus_url='http://10.105.191.163:30000/',
                                   query='collectd_cpufreq{cpufreq="0"}',
                                   starttime="2022-09-27T00:00:00Z",
                                   endtime="2022-09-28T00:00:00Z",
                                   step="1m")
```

### 3. Summary of the change 
- [x] Generate dataframe from Prometheus data.
- [x] If columns are not explicitly stated, `target_col` is automatically specified according to the Prometheus data, `id_col` is interpreted as only containing a single id and `extra_feature_col` is None.
- [x] Users can also customize columns.

### 4. How to test?
- [ ] Unit test
(Considering the inconvenience of obtaining data from Prometheus server, the API is only tested locally.)

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    _check_type(endtime, "endtime", (str, float))
+    _check_type(step, "step", (str, float))
+
+    # Generate dateframe
+    Pro_client = Prometheus(prometheus_url)
+    Pro_df = Pro_client.query_range(query, starttime, endtime, step, **kwargs)
+    df = pd.DataFrame(columns=Pro_df.columns.tolist())
+    df.insert(0, "datetime", Pro_df.index)
+    for col in Pro_df.columns.tolist():
+        df[col] = Pro_df[col].values
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py:170]: preferably, we don't use capital latter for variables (but we don't need to change it now)

File: python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                output_columns[col] = columns[col][0]  # id_col is str
+            else:
+                output_columns[col] = columns[col]
+        if output_columns[col] is not None:
+            if isinstance(output_columns[col], list):
+                output_col_list = output_col_list + output_columns[col]
+            else:
+                output_col_list.append(output_columns[col])
+
+    df = df.loc[:, output_col_list]
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py:192]: does `output_col_list` contains user input `extra_feature_col`?

File: python/chronos/src/bigdl/chronos/data/tsdataset.py [hunk 3]
```diff
@@ -213,6 +213,79 @@ def from_parquet(path,
...
+        df, df_columns = GetRangeDataframe(prometheus_url, query, starttime, endtime,
+                                           step, columns=columns, **kwargs)
+        return TSDataset.from_pandas(df,
+                                     dt_col=df_columns["dt_col"],
+                                     target_col=df_columns["target_col"],
+                                     id_col=df_columns["id_col"],
+                                     extra_feature_col=df_columns["extra_feature_col"],
+                                     with_split=with_split,
+                                     val_ratio=val_ratio,
+                                     test_ratio=test_ratio)
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/tsdataset.py:287]: we may add some white box check (not in this PR).

Comment (plusbang) [reply python/chronos/src/bigdl/chronos/data/tsdataset.py:287]: I think it's a good idea, we can try it after this PR

File: python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py (hunk 2 above)
Comment (plusbang) [reply python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py:192]: yes, the `output_col_list` is confirmed according to `output_columns`. `output_columns[extra_feature_col]` defaults to be None and will be updated if user specified `extra_feature_col`.

Comment (TheaperDeng) [reply python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py:197]: okay I see, yeah it should be correct

**Review by TheaperDeng — APPROVED** [review]

**Pull request merged**